src/main/java/com/systemmeltdown/meltdownlog/LogSession.java

```diff
@@ -0,0 +1,90 @@
+package com.systemmeltdown.meltdownlog;
+
+import com.systemmeltdown.meltdownlog.lib.NanoTimeReference;
+import com.systemmeltdown.meltdownlog.outputs.LogOutput;
+import com.systemmeltdown.meltdownlog.topics.LogTopic;
+import com.systemmeltdown.meltdownlog.topics.LogTopicRegistry;
+
+/**
+ * Represents a time-based session of logging. Each session is pre-configured
+ * and logs to its outputs after it's started until it is stoppped.
+ */
+public abstract class LogSession implements NanoTimeReference {
```

Comment:
Why is this abstract?

Reply:
The intent is to subclass a session with a specific implementation of a session with pre-configured outputs. (see the readme)